forest/src/daemon.rs

```diff
@@ -298,48 +265,101 @@ pub(super) async fn start(config: Config) {
         Arc::new(genesis),
         chain_muxer_tipset_sink,
         tipset_stream,
-        config.sync,
+        config.sync.clone(),
     )
     .expect("Instantiating the ChainMuxer must succeed");
     let bad_blocks = chain_muxer.bad_blocks_cloned();
     let sync_state = chain_muxer.sync_state_cloned();
-    let sync_task = task::spawn(chain_muxer);
+    services.push(task::spawn(async {
+        chain_muxer.await;
+    }));
 
     // Start services
-    let p2p_task = task::spawn(async {
-        p2p_service.run().await;
-    });
-    let rpc_task = if config.client.enable_rpc {
+    if config.client.enable_rpc {
         let keystore_rpc = Arc::clone(&keystore);
         let rpc_listen = TcpListener::bind(&config.client.rpc_address)
             .await
             .unwrap_or_else(|_| cli_error_and_die("could not bind to {rpc_address}", 1));
 
-        Some(task::spawn(async move {
+        let rpc_state_manager = Arc::clone(&state_manager);
+        let rpc_chain_store = Arc::clone(&chain_store);
+
+        services.push(task::spawn(async move {
             info!("JSON-RPC endpoint started at {}", config.client.rpc_address);
-            start_rpc::<_, _, FullVerifier, cns::FullConsensus>(
+            // XXX: The JSON error message are a nightmare to print.
```

Comment:
I'm not sure what does this comment refers to? do we need it?

Reply:
The line below silently ignores a json-rpc error. Ideally I would like to log it but there's no standard way of pretty-printing those errors.

Reply:
(The error was also silently ignored previously.)